core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/cordaInteropApp/flows/WriteExternalStateFlows.kt

```diff
@@ -126,25 +131,43 @@ class GetExternalStateByLinearId(
-                    val response = interopPayload.payload.toStringUtf8()
-                    println("response from remote: ${response}.\n")
+                    val payloadString = interopPayload.payload.toStringUtf8()
+                    println("response from remote: ${payloadString}.\n")
                     println("query address: ${interopPayload.address}.\n")
 
-                    var securityDomain = interopPayload.address.split("/")[1]
+                    val securityDomain = interopPayload.address.split("/")[1]
                     val proofStringPrefix = "Verified Proof: Endorsed by members: ["
-                    val proofStringSuffix = "] of security domain: " + securityDomain
-                    var orgList = ""
+                    val proofStringSuffix = "] of security domain: $securityDomain"
+                    var mspIdList = ""
                     fabricViewData.endorsementsList.map { endorsement ->
-                        val org = Identities.SerializedIdentity.parseFrom(endorsement.endorser).mspid
-                        if (orgList.length > 0) {
-                            orgList = orgList + ", "
+                        val mspId = Identities.SerializedIdentity.parseFrom(endorsement.endorser).mspid
+                        if (mspIdList.isNotEmpty()) {
+                            mspIdList += ", "
                         }
-                        orgList = orgList + org
+                        mspIdList += mspId
                     }
-                    var proof = proofStringPrefix + orgList + proofStringSuffix
-                    println("Proof: ${orgList}.\n")
+                    val proofMessage = proofStringPrefix + mspIdList + proofStringSuffix
+                    println("Proof Message: ${proofMessage}.\n")
+
+                    var signatureList: List<ProofSignature> = listOf()
 
-                    return Pair(response.toByteArray(), proof.toByteArray())
+                    fabricViewData.endorsementsList.map { endorsement ->
+                        val serializedIdentity = Identities.SerializedIdentity.parseFrom(endorsement.endorser)
+                        val mspId = serializedIdentity.mspid
+                        val certString = Base64.getEncoder().encodeToString(serializedIdentity.idBytes.toByteArray())
+                        val signature = Base64.getEncoder().encodeToString(endorsement.signature.toByteArray())
+                        val proofSignature = ProofSignature(mspId, certString, signature)
+                        signatureList = signatureList + proofSignature
+                    }
+
+                    val response = ExternalStateResponse(payloadString, signatureList, proofMessage)
+
+                    val mapper = jacksonObjectMapper()
+                    val responseJSONString = mapper.writeValueAsString(response)
+
+                    println("Return Proof: ${responseJSONString}.\n")
+
+                    return responseJSONString.toByteArray()
```

Comment:
I suggest we use a protobuf here (defined in `common/interop-protos/corda`) instead of a JSON as different CorDapps will need to understand this structure,